AppCenterCrashes/AppCenterCrashes/Internals/Util/MSCrashesCategory.m

```diff
@@ -0,0 +1,210 @@
+ *      UncaughtExceptionHandler.
+ *
+ *      Example:
+ *        - (IBAction)doExceptionCrash:(id)sender {
+ *          NSArray *array = [NSArray array];
+ *          [array objectAtIndex:23];
+ *        }
+ *
+ *
+ * Solution A:
+ *
+ *   Implement `NSExceptionHandler` and set the `ExceptionHandlingMask` to `NSLogAndHandleEveryExceptionMask`
+ *
+ *   Benefits:
+ *
+ *     1. Solves all of the above scenarios
+ *
+ *     2. Clean solution using a standard Cocoa System specifically meant for this purpose.
+ *
+ *     3. Safe. Doesn't use private API.
+ *
+ *   Problems:
+ *
+ *     1. To catch all exceptions the `NSExceptionHandlers` mask has to include `NSLogOtherExceptionMask` and
+ *        `NSHandleOtherExceptionMask`. But this will result in @catch blocks to be called after the exception
+ *        handler processed the exception and likely lets the app crash and create a crash report.
+ *        This makes the @catch block basically not working at all.
+ *
+ *     2. If anywhere in the app a custom `NSUncaughtExceptionHandler` will be registered, e.g. in a closed source
+ *        library the develop has to use, the complete mechanism will stop working
+ *
+ *     3. Not clear if this solves all scenarios there can be.
+ *
+ *     4. Requires to adjust PLCrashReporter not to register its `NSUncaughtExceptionHandler` which is not a good idea,
+ *        since it would require the `NSExceptionHandler` would catch *all* exceptions and that would cause
+ *        PLCrashReporter to stop all running threads every time an exception occurs even if will be handled right
+ *        away, e.g. by a system framework.
+ *
+ *
+ * Solution B:
+ *
+ *   Overwrite and extend specific methods of `NSApplication`. Can be implemented via subclassing NSApplication or
+ *   by using a category.
+ *
+ *   Benefits:
+ *
+ *     1. Solves scenarios 2 (by overwriting `reportException:`) and 3 (by overwriting `sendEvent:`)
+ *
+ *     2. Subclassing approach isn't enforcing the mechanism onto apps and let developers opt-in.
+ *        (Category approach would enforce it and rather be a problem of this soltuion.)
+ *
+ *     3. Safe. Doesn't use private API.
+ *
+ *  Problems:
+ *
+ *     1. Does not automatically solve scenario 1. Developer would have to put all that code into @try @catch blocks
+ *
+ *     2. Not a clean implementation, rather feels like a workaround.
+ *
+ *     3. Not clear if this solves all scenarios there can be.
+ *
+ *
+ * References:
+ *   https://developer.apple.com/library/mac/documentation/cocoa/Conceptual/Exceptions/Tasks/ControllingAppResponse.html#//apple_ref/doc/uid/20000473-BBCHGJIJ
+ *   http://stackoverflow.com/a/4199717/474794
+ *   http://stackoverflow.com/a/3419073/474794
+ *   http://macdevcenter.com/pub/a/mac/2007/07/31/understanding-exceptions-and-handlers-in-cocoa.html
+ *
+ */
+@implementation NSApplication (CrashException)
```

Comment:
I'd prefix the category and category methods names since this is a category over an Apple class.